src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -198,6 +203,10 @@ public function has($id)
             if (--$i && $id !== $lcId = strtolower($id)) {
                 $id = $lcId;
             } else {
+                if (isset($this->privates[$id])) {
+                    @trigger_error(sprintf('Checking for the existence of the private service "%s" is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0. An exception will be thrown instead in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
Why throw an exception in this case? If you check a private service for existence, the method should return `false`, right?